api/build/restart.go

```diff
@@ -105,6 +107,25 @@ func RestartBuild(c *gin.Context) {
 		return
 	}
 
+	// check to see if queue has reached configured capacity to allow restarts for pending builds
+	if b.GetStatus() == constants.StatusPending && ps.GetQueueRestartLimit() > 0 {
+		// check length of specified route for the build
+		queueLength, err := queue.FromContext(c).RouteLength(ctx, b.GetRoute())
+		if err != nil {
+			util.HandleError(c, http.StatusInternalServerError, fmt.Errorf("unable to get queue length for %s: %w", b.GetRoute(), err))
+
+			return
+		}
+
+		if queueLength >= int64(ps.GetQueueRestartLimit()) {
+			retErr := fmt.Errorf("unable to restart build %s: queue length %d exceeds configured limit %d. Please wait for the queue to decrease in size before retrying", entry, queueLength, ps.GetQueueRestartLimit())
```

Comment:
i like it, but always think it's strange to start capitalizing in the middle of error message. perhaps `;` or `,` so we can cheat and continue w/ lower case?

```suggestion
			retErr := fmt.Errorf("unable to restart build %s: queue length %d exceeds configured limit %d, please wait for the queue to decrease in size before retrying", entry, queueLength, ps.GetQueueRestartLimit())
```